py3dtilers/CityTiler/citym_waterbody.py

```diff
@@ -94,3 +94,56 @@ def sql_query_geometries(waterbodies_ids=None, split_surfaces=False):
                 "GROUP BY waterbody.id "
 
         return query
+
+    @staticmethod
+    def sql_query_geometries_with_texture_coordinates(water_bodies_ids=None):
+        """
+        param water_bodies_ids: a formatted list of (city)gml identifier corresponding to
+                            objects_type type objects whose geometries are sought.
+        :return: a string containing the right sql query that should be executed.
+        """
+        # cityobjects_ids contains ids of water_bodies
+        query = \
+            ("SELECT surface_geometry.id, "
+             "ST_AsBinary(ST_Multi(surface_geometry.geometry)) as geom, "
+             "ST_AsBinary(ST_Multi(ST_Translate(ST_Scale(textureparam.texture_coordinates, 1, -1), 0, 1))) as uvs, "
+             "tex_image_uri AS uri "
+             "FROM citydb.waterbody JOIN citydb.waterbod_to_waterbnd_srf "
+             "ON waterbody.id=waterbod_to_waterbnd_srf.waterbody_id "
+             "JOIN citydb.waterboundary_surface "
+             "ON waterbod_to_waterbnd_srf.waterboundary_surface_id=waterboundary_surface.id "
+             "JOIN citydb.surface_geometry "
+             "ON surface_geometry.root_id=waterboundary_surface.lod3_surface_id "
+             "JOIN citydb.textureparam "
+             "ON textureparam.surface_geometry_id=surface_geometry.id "
+             "JOIN citydb.surface_data "
+             "ON textureparam.surface_data_id=surface_data.id "
+             "JOIN citydb.tex_image "
+             "ON surface_data.tex_image_id=tex_image.id "
+             "WHERE waterbody.id IN " + water_bodies_ids)
+        return query
+
+    @staticmethod
+    def sql_query_centroid(id):
```

Comment:
What use of this method do we have ? Is it related only to CGML waterbody ? If not, a virtual method should be written on citym_cityobjects

Reply:
This method is used to find the centroid of the CityObject. When computing surfaces separately (with `--with_texture` or `--split_surface` flags), this method allows to use the centroid the parent CityObject (instead of the centroid of the surface) to create LODs or kd_trees.

This method is implemented in each class inheriting from citym_cityobjects.